File: NextGen-BM-BE/NextGen-BM-BE-Domain/Entities/Property.cs
namespace NextGen_BM_BE_Domain;

public class Property{
    public int PropertyID { get; set; }
    public int PropertyNumber { get; set; }
    public int BuildingID { get; set; }
    public double Size { get; set; }
    public int Floor { get; set; }
    public double SizeOfIdealParts { get; set; }
    public int PropertyTypeID { get; set; }
    public bool entranceIsExternal { get; set; }

}

File: NextGen-BM-FE/src/models/property.ts
export interface Property{
    propertyId: number;
    propertyNumber: number;
    buildingId: number;
    size: number;
    floor: number;
    sizeOfIdealParts: number;
    entranceIsExternal: boolean;
    payments: PropertyPayments[] | null;
    residentHistory: ResidentHistory[] | null;
}

export interface ResidentHistory{
    propertyResidentsId: number;
    firstName: string;
    lastName: string;
    residentTypeId: number;
    enterDate: Date;
    leaveDate: Date;
}

export interface PropertyPayments{
    paymentId: number;
    propertyId: number;
    amountOwed: number;
    dateOpened: Date;
    dueDate: Date;
    status?: string;
    paymentMethod?: string;
}

export interface PropertyExpense{
    propertyExpenseId: number;
    propertyExpenseTemplateId: number;
    responsibleRole: string;
    price:number;
    startDate: Date;
    endDate: Date;
    description: string;
}